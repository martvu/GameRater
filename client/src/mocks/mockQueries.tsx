import {
  ADD_FAVORITES,
  CREATE_REVIEW,
  REMOVE_FAVORITES,
  SIGN_IN_OR_CREATE_USER,
} from '@/lib/mutations';
import {
  GET_GAMES,
  GET_GAME,
  GET_FILTERS,
  GET_GAME_PLATFORMS,
  GET_SEARCH_SUGGESTIONS,
} from '@/lib/queries';

const getGamesMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: undefined,
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesMockedKeyword = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesSignedInMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '1',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            platforms: [1, 2],
            genres: [3, 4],
          },
          // Add more game objects as needed
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getFiltersMock = {
  request: {
    query: GET_FILTERS,
    variables: {},
  },
  result: {
    data: {
      getGenres: [
        { id: 1, name: 'Genre 1' },
        { id: 2, name: 'Genre 2' },
      ],
      getPlatforms: [
        { id: 1, name: 'Platform 1' },
        { id: 3, name: 'Platform 3' },
        { id: 4, name: 'PC (Microsoft Windows)' },
        { id: 2, name: 'Nintendo Switch' },
        { id: 5, name: 'PlayStation 5' },
        { id: 6, name: 'Xbox Series X|S' },
      ],
    },
  },
};

const getGamePlatformsMock = {
  request: {
    query: GET_GAME_PLATFORMS,
    variables: { id: '1' },
  },
  result: {
    data: {
      getGame: {
        name: 'Example Game',
        platforms: [
          {
            id: 3,
            name: 'Platform 3',
          },
          {
            id: 4,
            name: 'PC (Microsoft Windows)',
          },
        ],
      },
    },
  },
};

const signInMock = {
  request: {
    query: SIGN_IN_OR_CREATE_USER,
    variables: {
      userInput: {
        username: 'testuser',
      },
    },
  },
  result: {
    data: {
      signInOrCreateUser: {
        _id: '1',
        username: 'testuser',
        favorites: [],
        reviews: [],
      },
    },
  },
};

const getGameMock = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: '',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};

const getGameMock2 = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: 'testUser',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};
const addFavoritesMock = {
  request: {
    query: ADD_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      addFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [{ _id: '1' }],
      },
    },
  },
};

const removeFavoritesMock = {
  request: {
    query: REMOVE_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      removeFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [],
      },
    },
  },
};

const createReviewMock = {
  request: {
    query: CREATE_REVIEW,
    variables: {
      reviewInput: {
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        user: 'testUser',
        gameID: '1',
      },
    },
  },
  result: {
    data: {
      createReview: {
        _id: 'review1',
        user: 'testUser',
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        gameID: '1',
      },
    },
  },
};

const searchbarSuggestionsMock = {
  request: {
    query: GET_SEARCH_SUGGESTIONS,
    variables: {
      query: 'example',
    },
  },
  result: {
    data: {
      getSearchSuggestions: [
        {
          _id: '1',
          name: 'Example Game',
          cover_image_id: 'cover1',
        },
        {
          _id: '2',
          name: 'Example Game 2',
          cover_image_id: 'cover2',
        },
      ],
    },
  },
};

const allMocks = [
  createReviewMock,
  getGamePlatformsMock,
  getGamesMockedKeyword,
  addFavoritesMock,
  removeFavoritesMock,
  getGamesMock,
  getFiltersMock,
  signInMock,
  getGamesSignedInMock,
  getGameMock,
  getGameMock2,
  searchbarSuggestionsMock,
  // Add more mocks as needed for other queries
];

export {
  allMocks,
  getGamesMock,
  getGamesSignedInMock,
  getFiltersMock,
  signInMock,
};
