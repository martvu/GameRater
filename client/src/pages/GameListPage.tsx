import FilterModal from '@/components/FilterModal';
import Filters from '@/components/Filters';
import SortBy from '@/components/SortBy';
import GamesList from '@/components/GamesList';
import withLayout from '@/lib/withLayout';
import UserFilters from '@/components/UserFilters';

function BaseGameListPage() {
  return (
    <section className="grid flex-grow grid-cols-1 justify-center md:grid-cols-[auto,1fr] md:justify-normal">
      {/* Filter section visible on larger screens */}
      <aside className="fixed top-16 z-30 hidden h-[calc(100vh-3.5rem)] w-full shrink-0 px-2 pt-3 md:sticky md:block ">
        <Filters />
      </aside>
      {/* Games list section */}
      <section className="pl-2 pr-2 pt-4 sm:pr-4">
        <div className="z-10 flex items-center justify-between gap-1 bg-background pb-2">
          {/* Filter modal visible on smaller screens */}
          <div className="block md:hidden">
            <FilterModal />
          </div>
          <UserFilters />
          <SortBy />
        </div>
        <GamesList />
      </section>
    </section>
  );
}

const GameListPage = withLayout(BaseGameListPage);

export default GameListPage;
