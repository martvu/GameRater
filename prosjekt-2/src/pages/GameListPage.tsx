import Filters from "@/components/Filters";
import GameList from "@/components/GameList";
import Nav from "@/components/Nav";

export default function GameListPage() {
  return (
    <div className="flex flex-col">
      <Nav />
      <div className="flex md:flex-row flex-col">
        <Filters/>
        <GameList />
      </div>
      
    </div>
  )
}
