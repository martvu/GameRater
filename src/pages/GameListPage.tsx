import FilterModal from '@/components/FilterModal';
import Filters from '@/components/Filters';
import { GameCard } from '@/components/GameCard';
import Botw from '../assets/botw.jpeg';
import { gameData } from '@/components/GameData';
import SortBy from '@/components/SortBy';
import Pagination from '@/components/Pagination';
import { useState } from 'react';
import withLayout from '@/lib/withLayout';

function BaseGameListPage() {
  const [currentPage, setCurrentPage] = useState(1);
  const gamesPerPage = 12;

  return (
    <div className="grid flex-grow grid-cols-1 justify-center overflow-auto md:grid-cols-[auto,1fr] md:justify-normal">
      <div className="ml-4 hidden md:block">
        <Filters />
      </div>
      <div className="overflow-x-hidden px-8">
        <div className="sticky top-0 z-10 flex justify-between bg-background py-1 pb-4 md:justify-end">
          <div className="block md:hidden">
            <FilterModal />
          </div>
          <SortBy />
        </div>
        <div className="grid grid-cols-[repeat(auto-fill,minmax(240px,1fr))] justify-center gap-4">
          {gameData
            .slice((currentPage - 1) * gamesPerPage, currentPage * gamesPerPage)
            .map((game, index) => (
              <div
                className="m-1 flex justify-center md:justify-normal"
                key={game.title}
              >
                <GameCard
                  id={index} // or some  unique identifier
                  title={game.title}
                  description={game.description}
                  image={game.image ? game.image : Botw}
                  rating={game.rating}
                />
              </div>
            ))}
        </div>
        <div className="mt-2 flex w-full justify-center">
          <Pagination
            currentPage={currentPage}
            setCurrentPage={setCurrentPage}
            gamesPerPage={gamesPerPage}
            gameData={gameData}
          />
        </div>
      </div>
    </div>
  );
}

const GameListPage = withLayout(BaseGameListPage);
export default GameListPage;
