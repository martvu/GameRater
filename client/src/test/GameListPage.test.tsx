import { act, screen, waitFor } from '@testing-library/react';
import GameListPage from '@/pages/GameListPage';
import { vi, describe, it } from 'vitest';
import userEvent from '@testing-library/user-event';
import { allMocks as mocks } from '../mocks/mockQueries';
import { testPageRender } from './test-utils';

beforeAll(() => {
  // Mock `window.scrollTo` to accept any arguments
  window.scrollTo = vi.fn(() => {});
});

// Mock 'react-router-dom'
vi.mock('react-router-dom', async () => {
  const actualModule =
    await vi.importActual<typeof import('react-router-dom')>(
      'react-router-dom'
    );

  return {
    ...actualModule,
    useParams: vi.fn(() => ({ keyword: 'mockedKeyword' })),
  };
});

describe('GameListPage Component', () => {
  it('matches the snapshot', async () => {
    const { asFragment } = testPageRender(<GameListPage />, { mocks });

    // Wait for any asynchronous operations to complete
    await waitFor(() => {
      expect(screen.getByText('Example Game')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
  it('renders the GamesList component with mocked GraphQL data', async () => {
    testPageRender(<GameListPage />, { mocks });
    screen.debug();
    // Wait for the mocked data to be displayed
    await waitFor(() => {
      expect(screen.getByText('Example Game')).toBeInTheDocument();
      expect(screen.getByText('1 results')).toBeInTheDocument();
      expect(screen.getByText('4.5')).toBeInTheDocument();
    });
  });

 it('signs in and renders correctly', async () => {
    testPageRender(<GameListPage />, { mocks });
    await userEvent.click(screen.getByText('Sign In'));
    await userEvent.type(screen.getByLabelText('Username'), 'testuser{enter}');
    await userEvent.click(screen.getByText('T'));
  
    await waitFor(() => {
      expect(screen.getByText('testuser')).toBeInTheDocument();
      expect(screen.getByText('Sign out')).toBeInTheDocument();
      expect(screen.getByText('T')).toBeInTheDocument();
      expect(screen.getByText('Example Game')).toBeInTheDocument();
      expect(screen.getByText('Favorites')).toBeInTheDocument();
      expect(screen.getByText('Reviewed')).toBeInTheDocument();
    });
  }); 


  // NOT WORKING
  /* it('gets sign in prompt when trying to favorite a game', async () => {
    testPageRender(<GameListPage />, { mocks });
    await userEvent.click(screen.getAllByTestId('favorite-btn')[0]);
    await act(async () => {
      
      screen.debug();
      expect(screen.getByText('Sign In')).toBeInTheDocument();
      
  
      // Wait for expected outcome after the click
      await waitFor(() => {
        expect(screen.getByText('Please sign in to favorite this game!')).toBeInTheDocument();
      });
    });
  });  */
  
});
