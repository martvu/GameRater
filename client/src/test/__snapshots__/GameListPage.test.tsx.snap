// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameListPage Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen flex-col"
  >
    <header
      class="fixed top-0 z-40 flex h-16 w-[calc(100vw-0.75rem)] items-center bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/80"
    >
      <nav
        class="flex w-full items-center justify-between"
      >
        <a
          aria-label="Return to Home Page"
          data-testid="logo-btn"
          href="/"
        >
          <div
            class="w-28 items-center justify-between gap-2 px-2 flex"
          >
            <button
              class="inline-flex items-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 justify-start hover:text-gray-400 h-10 px-4 py-2 hover:opacity-80"
            >
              <span
                class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
              >
                <span
                  class="flex h-full w-full items-center justify-center rounded-full bg-muted"
                >
                  CN
                </span>
              </span>
              <h1
                class="ml-1 text-xl font-bold tracking-wide text-primary"
              >
                GameRater
              </h1>
            </button>
          </div>
        </a>
        <div
          class="w-full max-w-sm flex-grow items-center justify-center space-x-2 md:flex hidden md:flex"
        >
          <form
            class="flex"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
              for="search"
            />
            <div
              class="relative w-[200px] md:w-[300px] lg:w-[400px]"
            >
              <input
                class="flex h-10 border border-input px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 w-full rounded-xl bg-muted pr-24 shadow-inner md:w-full lg:w-full"
                data-testid="search-input"
                placeholder="mockedKeyword"
                type="text"
                value=""
              />
              <button
                aria-label="Search"
                class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-transparent text-secondary-foreground hover:text-muted-foreground transition duration-300 ease-in-out h-10 py-2 absolute right-0 top-0 rounded-l-none rounded-r-xl px-3 active:opacity-90"
                type="submit"
              >
                <svg
                  class="lucide lucide-search"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <path
                    d="m21 21-4.3-4.3"
                  />
                </svg>
              </button>
            </div>
          </form>
        </div>
        <div
          class="flex-shrink-0 items-center md:gap-2 flex"
        >
          <button
            class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-2xl md:hidden"
          >
            <svg
              class="lucide lucide-search"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <path
                d="m21 21-4.3-4.3"
              />
            </svg>
          </button>
          <div>
            <button
              aria-controls="radix-:r0:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              data-state="closed"
              data-testid="modal-sign-in-button"
              type="button"
            >
              Sign In
            </button>
          </div>
          <div
            class="mr-1"
          >
            <button
              class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-2xl"
            >
              <svg
                class="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <path
                  d="M12 2v2"
                />
                <path
                  d="M12 20v2"
                />
                <path
                  d="m4.93 4.93 1.41 1.41"
                />
                <path
                  d="m17.66 17.66 1.41 1.41"
                />
                <path
                  d="M2 12h2"
                />
                <path
                  d="M20 12h2"
                />
                <path
                  d="m6.34 17.66-1.41 1.41"
                />
                <path
                  d="m19.07 4.93-1.41 1.41"
                />
              </svg>
              <svg
                class="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Toggle theme
              </span>
            </button>
          </div>
        </div>
      </nav>
    </header>
    <div
      class="flex-grow"
    >
      <main
        class="grid flex-grow grid-cols-1 justify-center pt-16 md:grid-cols-[auto,1fr] md:justify-normal"
      >
        <div
          class="hidden px-2 md:block"
        >
          <aside
            class="fixed top-16 z-30 hidden h-[calc(100vh-3.5rem)] w-full shrink-0 pt-3 md:sticky md:block "
          >
            <div
              class="flex h-full flex-col items-start overflow-hidden text-left transition-all duration-300 ease-in-out w-[240px]"
            >
              <div
                class="relative overflow-hidden h-full pb-4 pr-2"
                dir="ltr"
                style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
              >
                <style>
                  [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
                </style>
                <div
                  class="h-full w-full rounded-[inherit]"
                  data-radix-scroll-area-viewport=""
                  style="overflow-x: hidden; overflow-y: scroll;"
                >
                  <div
                    style="min-width: 100%; display: table;"
                  >
                    <div
                      class="pb-4 pl-5 md:pb-32"
                    >
                      <h1
                        class="mt-4 text-left text-xl font-bold tracking-wider text-foreground "
                      >
                        Filter By
                      </h1>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <button
              aria-label="toggle filters"
              class="items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 fixed bottom-4 left-4 z-50 m-1 hidden md:flex"
            >
              <svg
                class="lucide lucide-chevron-left"
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15 18-6-6 6-6"
                />
              </svg>
              <span
                class="ml-2"
              >
                Hide filters
              </span>
            </button>
          </aside>
        </div>
        <div
          class="pl-2 pr-2 pt-4 sm:pr-4"
        >
          <div
            class="z-10 flex items-center justify-between gap-1 bg-background pb-2"
          >
            <div
              class="block md:hidden"
            >
              <button
                aria-controls="radix-:r3:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[150px]"
                data-state="closed"
                type="button"
              >
                Filters
              </button>
            </div>
            <div
              class="flex gap-2"
            />
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r6:"
              aria-expanded="false"
              class="flex h-10 items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 w-[150px] hover:bg-accent"
              data-placeholder=""
              data-state="closed"
              data-testid="sort-by-select"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                Sort By
              </span>
              <svg
                aria-hidden="true"
                class="h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="overflow-y-auto"
          >
            <main
              class="mt-1"
            >
              <div
                class="flex items-center gap-2 pb-4 text-muted-foreground opacity-80"
              >
                <p
                  class="w-[100px] text-left text-sm"
                >
                  1 results
                </p>
              </div>
              <ul
                class="grid grid-cols-[repeat(auto-fill,minmax(175px,1fr))] justify-center gap-1 sm:grid-cols-[repeat(auto-fill,minmax(260px,1fr))] sm:gap-4"
              >
                <li
                  class="m-1 flex justify-center"
                >
                  <div
                    class="rounded-lg border bg-card text-card-foreground shadow-lg h-[310px] min-w-[175px] max-w-[262px] overflow-visible p-0 duration-300 hover:scale-105 sm:h-[410px] sm:min-w-[262px]"
                  >
                    <a
                      aria-label="Link to Example Game detail page"
                      data-testid="game-card-link"
                      href="/game/1"
                    >
                      <div
                        class="flex flex-col h-[235px] w-[176px] overflow-hidden rounded-t-lg sm:mb-2 sm:h-[320px] sm:w-[260px]"
                      >
                        <div
                          class="flex flex-1 items-center justify-center"
                        >
                          <div
                            class="relative h-full w-full object-cover"
                          >
                            <div
                              class="absolute inset-0 z-10 flex items-center justify-center"
                            >
                              <img
                                alt="Example Game"
                                class="w-1/2 object-cover opacity-100 transition-opacity duration-500 ease-in-out"
                                src="/project2/src/assets/img-fallback.svg"
                              />
                            </div>
                            <img
                              alt="Example Game"
                              class="absolute inset-0 h-full w-full object-cover transition-opacity duration-500 ease-in-out z-0 opacity-0"
                              loading="lazy"
                              src="https://images.igdb.com/igdb/image/upload/t_cover_big/cover1.jpg"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="pt-0 relative px-3 text-start"
                      >
                        <h3
                          class="font-semibold tracking-tight line-clamp-1 max-w-[150px] py-1 text-sm duration-300 hover:opacity-50 sm:my-1 sm:max-w-[240px] sm:text-lg"
                          data-state="closed"
                        >
                          Example Game
                        </h3>
                      </div>
                    </a>
                    <div
                      class="flex items-center pt-0 mt-auto h-[40px] w-full pl-3 pr-2 sm:pb-2"
                    >
                      <div
                        class="relative flex h-full w-full items-center"
                      >
                        <div
                          class="absolute left-0 flex items-center"
                        >
                          <svg
                            class="mr-1 h-5 fill-yellow-400 text-yellow-400"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <polygon
                              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                            />
                          </svg>
                          <p>
                            4.5
                          </p>
                        </div>
                        <div
                          class="absolute right-10 flex items-center gap-2"
                        >
                          <div
                            aria-label="metascore"
                            class="flex items-center justify-center rounded-md border border-white text-sm text-white bg-green-600 h-7 w-7 px-1"
                          >
                            85
                          </div>
                        </div>
                        <div
                          class="absolute right-0"
                        >
                          <button
                            aria-label="Favorite game button"
                            class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-full z-10"
                            data-testid="favorite-btn"
                          >
                            <svg
                              class="none"
                              data-testid="heart-icon"
                              fill="none"
                              height="24"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
              <div
                class="my-2 flex w-full justify-center"
              />
            </main>
          </div>
        </div>
      </main>
    </div>
    <footer
      class="mt-5 h-16 w-full border-t text-slate-400"
    >
      <div
        class="container mx-auto px-6 pt-5 text-left text-sm"
      >
        Copyright 2023 GameRater
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
